src/components/v2/ValueUpdate/index.tsx

```diff
@@ -6,32 +6,24 @@ import { formatCentsToReadableValue } from 'utilities/common';
 import { Icon } from '../Icon';
 import { useStyles } from './styles';
 
-interface INumberValueProps {
-  original: number | undefined;
-  update: number | undefined;
-  format?: (value: number | undefined) => string;
-}
-
-interface IBigNumberValueProps {
-  original: BigNumber | undefined;
-  update: BigNumber | undefined;
-  format?: (value: BigNumber | undefined) => string;
-}
-
-interface ValueUpdateCommonProps {
+export interface IValueUpdateProps<T> {
+  original: T;
+  update: T;
+  format?: (value: T) => string;
   className?: string;
   positiveDirection?: 'asc' | 'desc';
 }
 
-export type ValueUpdateProps = ValueUpdateCommonProps & (INumberValueProps | IBigNumberValueProps);
-
-export const ValueUpdate: React.FC<ValueUpdateProps> = ({
+export function ValueUpdate<T>({
```

Comment:
```suggestion
export const ValueUpdate = <T,>({
```

Reply:
TypeScirpt doesn't seem to like that. I haven't seen that notation before 
<img width="1354" alt="Screen Shot 2022-05-05 at 14 17 07" src="https://user-images.githubusercontent.com/44675210/166931112-15675146-9033-4a24-9ac3-da3d3cc59f5a.png">

Reply:
Interesting. I didn't receive any errors when I tested it. That syntax generally works but the compiler can get tripped up in `.tsx` files